components/camel-tracing/src/main/java/org/apache/camel/tracing/ActiveSpanManager.java

```diff
@@ -56,6 +56,13 @@ public static SpanAdapter getSpan(Exchange exchange) {
      * @param span     The span
      */
     public static void activate(Exchange exchange, SpanAdapter span) {
+        if (exchange.getProperty(ExchangePropertyKey.CLOSE_CLIENT_SCOPE, Boolean.FALSE, Boolean.class)) {
+            //Check if we need to close the CLIENT scope created by
+            //DirectProducer in async mode before we create a new INTERNAL scope
+            //for the next DirectConsumer
+            endScope(exchange);
+            exchange.setProperty(ExchangePropertyKey.CLOSE_CLIENT_SCOPE, Boolean.FALSE);
```

Comment:
Its likely not needed to set this as the default value above is already false

Reply:
Hi @davsclaus ,

We run into this piece of code only when property ExchangePropertyKey.CLOSE_CLIENT_SCOPE is true, and we need to switch it as false to avoid doing endScope(exchange) again.

Thanks for review!
Freemna